process/block/postprocess/intermediateResults.go

```diff
@@ -235,6 +241,31 @@ func (irp *intermediateResultsProcessor) AddIntermediateTransactions(txs []data.
 	return nil
 }
 
+func (irp *intermediateResultsProcessor) checkSmartContractResultIntegrity(scr *smartContractResult.SmartContractResult) error {
+	if len(scr.RcvAddr) == 0 {
```

Comment:
we could have made InterceptedUnsignedTransaction.integrity() exported function without pointer receiver (with the SmartContractResult parameter) and used it here.

Reply:
I do not think it will work - maybe we will have some cyclic imports.